app/src/main/java/eu/faircode/netguard/Util.java

```diff
@@ -279,15 +279,23 @@ public static boolean hasPhoneStatePermission(Context context) {
     }
 
     public static List<String> getDefaultDNS(Context context) {
-        String dns1 = jni_getprop("net.dns1");
-        String dns2 = jni_getprop("net.dns2");
+        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
+        String vpnDns = prefs.getString("dns", null);
+        boolean ignoresysdns = prefs.getBoolean("ignoresysdns", false);
         List<String> listDns = new ArrayList<>();
-        if (!TextUtils.isEmpty(dns1))
-            listDns.add(dns1);
-        if (!TextUtils.isEmpty(dns2))
-            listDns.add(dns2);
-        if (TextUtils.isEmpty(dns1) && TextUtils.isEmpty(dns2))
-            listDns.add("8.8.8.8");
+        if (!ignoresysdns) {
```

Comment:
It is best practice not to negate condition when there is no need to (when there is also an else).